### PR TITLE
GraphQL: Add mutations for managing gift registrants

### LINKED ISSUE #8603: GraphQL: Add mutations for managing gift registry registrants
# Community engineering issue

Type of Doc:

-  [x] New topic
-  [ ] Topic Update, Devdoc link:
-  [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/28586

Magento 2 GitHub PR: https://github.com/magento/partners-magento2ee/pull/280

Magento 2 version:

-  [x] 2.4.3

Notifications of Changes:

-  [ ] Changes to Admin interface?
-  [ ] New software requirements?
-  [x] New REST endpoints or updated GraphQL query syntax?
-  [ ] New command line option?

## Content

Add new topics that cover the following mutations

addGiftRegistryRegistrants
removeGiftRegistryRegistrants
updateGiftRegistryRegistrants

Acceptance Criteria:

* The PR must be submitted from a local branch off of the devdocs `gift-registry` branch
* Each topic must contain an example that uses the gift registry provided with the Luma sample data
* The GraphQL TOC file must be updated to include the new topics.

## Additional information

See #8600 if you're unsure about file locations, naming conventions, standards, etc.

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) adds the three mutations that manage gift registry registrants. It fixes #8603 

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

### CODE REVIEW

**Comment by keharper** [comment]
running tests

File: src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+group: graphql
+title: addGiftRegistryRegistrants mutation
+
+---
+The `addGiftRegistryRegistrants` mutation add a registrant to the specified gift registry.
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md:6]:
```suggestion
The `addGiftRegistryRegistrants` mutation adds a registrant to the specified gift registry.
```

File: src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+The following example adds a registrant to the specified gift registry.
+
+**Request:**
+
+```graphql
+mutation {
+  addGiftRegistryRegistrants (
+    giftRegistryUid: "W9YcRai9JmzGglqP3p0USodTTM3BmjjY", 
+    registrants: {
+        firstname: "Monica"
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md:34]: Would suggest using a name like "Jane Doe" as "Monica Resendez" is a real person.

File: src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+{
+  "data": {
+    "addGiftRegistryRegistrants": {
+      "gift_registry": {
+        "uid": "W9YcRai9JmzGglqP3p0USodTTM3BmjjY",
+        "event_name": "Theo's 45th Birthday",
+        "registrants": [
+          {
+            "uid": "Mg==",
+            "firstname": "Stacey",
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/add-gift-registry-registrants.md:64]: Would suggest using a name like "Jane Doe" & "John Doe" as "Monica Resendez" is a real person.

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 4]
```diff
@@ -52,8 +52,8 @@ mutation {
       registrants: [
         {
           firstname: "Julie"
-          lastname: "Mao"
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:55]: Would advise more generic names

File: src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
+---
+group: graphql
+title: removeGiftRegistryRegistrants mutation
+
+---
+The `removeGiftRegistryRegistrants` mutation .
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md:6]: You also haven't described what this mutation does - like you have on others.
```suggestion
The `removeGiftRegistryRegistrants` mutation.
```

File: src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        email
+      }
+    }
+  }
+}
+```
+
+**Response:**
+
+```json
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md:50]: Would advise more generic names

File: src/guides/v2.4/graphql/mutations/update-gift-registry-registrants.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        email
+      }
+    }
+  }
+}
+```
+
+**Response:**
+
+```json
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/update-gift-registry-registrants.md:54]: Would advise more generic names

File: src/guides/v2.4/graphql/mutations/update-gift-registry-registrants.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+`code` | ID! | A unique key for an additional attribute of the event
+`value` | String! | A corresponding value for the code
+
+## Output attributes
+
+The `UpdateGiftRegistryRegistrantsOutput` output object contains the following attribute.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`gift_registry` | [GiftRegistry](#GiftRegistry) | The gift registry after adding registrants
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/update-gift-registry-registrants.md:116]: Should this not say "after updating registrants" ?

File: src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+`giftRegistryRegistrantUid` | ID! | The unique ID of a `giftRegistryRegistrant` object
+`registrantsUid` | [ID!]! | An array of registrant IDs to remove
+
+## Output attributes
+
+The `RemoveGiftRegistryRegistrantsOutput` output object contains the following attribute.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`gift_registry` | [GiftRegistry](#GiftRegistry) | The gift registry after adding registrants
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/mutations/remove-gift-registry-registrants.md:85]: Should this not say "after removing registrants" ?

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice new topic - few corrections and suggestions made.

**Comment by keharper** [comment]
running tests

**Pull request merged**